### PR TITLE
Add a texture image related tests generator

### PR DESCRIPTION
**Pull request by zhenyao**
plus all the generated tests

### CODE REVIEW

**Comment by toji** [comment]
LGTM! This is definitely a more sane approach than duplicating all those variations by hand.

File: sdk/tests/py/tex_image_test_generator.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  {'internal_format': 'RG8UI', 'format': 'RG_INTEGER', 'type': 'UNSIGNED_BYTE' },
+  {'internal_format': 'RGB8', 'format': 'RGB', 'type': 'UNSIGNED_BYTE' },
+  {'internal_format': 'SRGB8', 'format': 'RGB', 'type': 'UNSIGNED_BYTE' },
+  {'internal_format': 'RGB565', 'format': 'RGB', 'type': 'UNSIGNED_BYTE' },
+  {'internal_format': 'RGB565', 'format': 'RGB', 'type': 'UNSIGNED_SHORT_5_6_5' },
+  {'internal_format': 'R11F_G11F_B10F', 'format': 'RGB', 'type': 'UNSIGNED_INT_10F_11F_11F_REV' },
+  {'internal_format': 'R11F_G11F_B10F', 'format': 'RGB', 'type': 'HALF_FLOAT' },
+  {'internal_format': 'R11F_G11F_B10F', 'format': 'RGB', 'type': 'FLOAT' },
+  {'internal_format': 'RRG9E5', 'format': 'RGB', 'type': 'HALF_FLOAT' },
+  {'internal_format': 'RRG9E5', 'format': 'RGB', 'type': 'FLOAT' },
```
Comment (kenrussell) [inline sdk/tests/py/tex_image_test_generator.py]: Typo here and above: should be RGB9_E5?

**Comment by zhenyao** [comment]
Thanks.  I'll fix it in a followup CL.

**Comment by kenrussell** [comment]
Bravo. Excellent work overall. One comment.

Do you think it's worth re-generating the WebGL 1.0 tests using the same script?

**Comment by zhenyao** [comment]
Yes, I am working on it at the moment.

**Pull request merged**